### PR TITLE
get and set our standardised remember me data

### PR DESCRIPTION
**Pull request by hxpaul**
#### What does this PR do? (please provide any background)

This is a new repo, abstracting the "previous search" saved data from the old wizard so a) it can be used in the new wizard and b) anywhere else.
We want to save the url a customer went to (or will be going to) at some point, along with some other data. Specifically when we redirect to trip controller from the last stage of the wizard we want to record this, plus the product they have chosen and the date of their trip. Then we want to get this back again later.
https://github.com/holidayextras/tripapp-product-parking/issues/21
#### What tests does this PR have?

Tests + ci in this pr
#### How can this be tested?

Does the code looks like it will do what we need?
#### What gif best describes how you feel about this work?

---
#### Developer Definition of Done/Quality Checklist (for PR author to complete BEFORE code review):
- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md) and I'm happy for this to be reviewed.
#### Software Engineer or Developer review:
- [x] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [x] I’ve checked for coding anti-patterns.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked all the tests are passing.
#### Software Engineer or project guru review:
- [x] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [x] I’ve checked for coding anti-patterns.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked all the tests are passing.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,104 @@
+{
```
Comment (hx-markterry) [inline .eslintrc]: can we remove this please

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  key: 'rememberMe'
+};
+
+rememberMe.set = function( params ) {
+  var defaults = {
+    url: location.href
+  };
+  var data = _.extend( defaults, params );
+  try {
+    localStorage[rememberMe.key] = JSON.stringify( data );
```
Comment (hx-markterry) [inline src/index.js:17]: Can we use an existing module like this one here instead: https://www.npmjs.com/package/putainde-localstorage

File: package.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  "devDependencies": {
+    "chai": "^3.4.1",
+    "istanbul": "^0.3.22",
+    "make-up": "^5.4.0",
+    "mocha": "^2.3.3",
+    "sinon": "^1.17.2"
+  },
+  "dependencies": {
+    "lodash.merge": "^3.3.2",
+    "tracker": "git@github.com:holidayextras/tracker.git"
```
Comment (shackpank) [inline package.json]: I don't think there's a reasonable way to have a public repo depend on a private one. It makes it impossible for "outsiders" to use, or contribute to, the module, and hooking up functional CI to it introduces security holes.

By the looks of things we only need the `tracker` module for its `error` method. Error handlers could be moved into the module and defaulted to a no-op or to a console warning. As long as we provide a way for it to be injected/overridden, we can still log those events to our (or anyone else's) internal data stores / GA accounts / Bugsnag accounts / whatever - it just has to be done by whatever repo this depends on, not directly by this one.

File: .eslintrc (hunk 1 above)
Comment (hxpaul) [inline .eslintrc]: oops, done

File: package.json (hunk 3 above)
Comment (hxpaul) [inline package.json]: thanks, have changed this to pass in an optional error callback and all is good

File: src/index.js (hunk 2 above)
Comment (hxpaul) [inline src/index.js:17]: As discussed I don't think that one is well enough supported, and we want a default format of data to make this easy to use.

**Comment by cameronviner** [comment]
Taking DEV

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+'use strict';
+var _ = { extend: require('lodash.merge') };
+
+var rememberMe = module.exports = {
+  key: 'rememberMe'
+};
+
+rememberMe.set = function( params, errorCallback ) {
+  var defaults = {
+    url: location.href
```
Comment (cameronviner) [inline src/index.js:10]: If we are trying to make this generic should we set this as a default rather than requiring the user to pass it in?

**Comment by cameronviner** [comment]
This looks awesome, only one minor comment that isn't a blocker just a question. :+1:

File: src/index.js (hunk 4 above)
Comment (hxpaul) [inline src/index.js:10]: it is the default, but you can override it... I thought it was something we'd always want to save, what url you are on, but in my very first use of this I needed to save the url you are about to redirect to instead...

**Comment by hx-markterry** [comment]
just putting this out there, could this be implemented as a react mixin instead that stores a component's value to local storage when changed?

**Comment by hx-markterry** [comment]
This be using a defined search model which will allow us to sync to local storage, this can then be used in other projects.

**Comment by hxpaul** [comment]
it's not meant as persistent storage for a component (though it could do that), my first use case is to save the search data url when we go to availability, but retrieve it again when building the destinations or airlines page so we can customise the results, I'll show you.
We could write a mixin which uses this if we needed it.

**Comment by hx-markterry** [comment]
ok with me :8ball:

**Pull request merged**